File: TypeScript/14. Longest Common Prefix.ts
// 14. Longest Common Prefix
// Level: Easy
// https://leetcode.com/problems/longest-common-prefix/

function longestCommonPrefix(strs: string[]): string {
    let result = strs[0];
    let minLength = result.length;

    for(let i = 1; i < strs.length; i++){
        minLength = Math.min(minLength, strs[i].length);
        for(let j = 0; j < minLength; j++){
            if(result[j] === strs[i][j]){
                continue;
            }
            minLength = Math.min(minLength, j);
            break;
        }
    }
    return result.substring(0, minLength);
};

File: CSharp/14. Longest Common Prefix.cs
// 14. Longest Common Prefix
// Level: Easy
// https://leetcode.com/problems/longest-common-prefix/

public class Solution
{
  public string LongestCommonPrefix(string[] strs)
  {
    var result = strs[0];
    var minLength = result.Length;

    for (var i = 1; i < strs.Length; i++)
    {
      minLength = Math.Min(minLength, strs[i].Length);
      for (var j = 0; j < minLength; j++)
      {
        if (result[j] == strs[i][j])
        {
          continue;
        }
        minLength = Math.Min(minLength, j);
        break;
      }
    }
    return result.Substring(0, minLength);
  }
}